collectivo/extensions/shifts/composables/shifts.ts

```diff
@@ -0,0 +1,300 @@
+
+  const start = DateTime.fromISO(
+    `${dateString}T${shift.shifts_from_time}Z`,
+  ).toUTC();
+
+  const end = DateTime.fromISO(
+    `${dateString}T${shift.shifts_to_time}Z`,
+  ).toUTC();
+
+  return {
+    shift: shift,
+    start: start,
+    end: end,
+    shiftRule: shiftRule,
+    slots: slotRules?.length ?? 0,
+    openSlots: openSlots,
+  };
+};
+
+// Create a RRule object for a shift
+// Shifts without end date run forever
+// Shifts without repetition run once
+// Dates are with T=00:00:00 UTC
+export const shiftToRRule = (shift: ShiftsShift): RRule => {
+  return new RRule({
+    freq: RRule.DAILY,
+    interval: shift.shifts_repeats_every,
+    count: shift.shifts_repeats_every ? null : 1,
+    dtstart: new Date(shift.shifts_from),
+    until: shift.shifts_to ? new Date(shift.shifts_to) : null,
+  });
+};
+
+// SlotRrule is a RRuleSet that shows only free occurences
+// Occurences with existing assignments are excluded
+export const slotToRrule = (
+  shift: ShiftsShift,
+  shiftRule: RRule,
+  assignments: ShiftsAssignment[],
+  absences: ShiftsAbsence[],
+): RRule => {
+  const slotRule = new RRuleSet();
+  slotRule.rrule(shiftRule);
+
+  for (const assignment of assignments) {
+    const assignmentRule = new RRuleSet();
+
+    assignmentRule.rrule(
+      new RRule({
+        freq: RRule.DAILY,
+        interval: shift.shifts_repeats_every,
+        dtstart: shiftRule.after(new Date(assignment.shifts_from), true),
+        until: assignment.shifts_to
+          ? shiftRule.before(new Date(assignment.shifts_to), true)
+          : null,
+      }),
+    );
+
+    const filteredAbsences = absences.filter(
+      (absence) =>
+        absence.shifts_assignment == assignment.id ||
+        absence.shifts_assignment == null,
+    );
+
+    for (const absence of filteredAbsences) {
+      const absenceRule = new RRule({
+        freq: RRule.DAILY,
+        interval: shift.shifts_repeats_every,
+        dtstart: shiftRule.after(new Date(absence.shifts_from), true),
+        until: shiftRule.before(new Date(absence.shifts_to), true),
+      });
+
+      assignmentRule.exrule(absenceRule);
+    }
+
+    // Exclude assignment from slotRules
+    slotRule.exrule(assignmentRule);
+  }
+
+  return slotRule;
+};
+
+export const isShiftDurationModelActive = (
+  durationModel: { shifts_from: string; shifts_to?: string },
+  atDate?: DateTime,
+): boolean => {
+  return isFromToActive(
+    DateTime.fromISO(durationModel.shifts_from),
+    durationModel.shifts_to
+      ? DateTime.fromISO(durationModel.shifts_to)
+      : undefined,
+    atDate,
+    true,
+  );
+};
+
+export const isFromToActive = (
+  from: DateTime,
+  to?: DateTime,
+  atDate?: DateTime,
+  dateOnly = true,
+): boolean => {
+  if (!atDate) {
+    atDate = DateTime.now();
+  }
+
+  if (dateOnly) {
+    from = from.startOf("day");
+    to = to?.endOf("day");
+  }
+
+  if (from > atDate) {
+    return false;
+  }
+
+  return !(to && to < atDate);
+};
```

Comment:
I thought I had removed those two since they are now under `utils`, I guess they got re-added by mistake when merging? 
`pnpm dev` gives warnings about them being defined twice.